### PR TITLE
[Azure] Add cis azure 9.4

### PR DESCRIPTION
**Pull request by orouz**
### Summary of your changes

adds CIS Azure 9.4 
Screenshot 2023-10-02 at 13 12 22

### Related Issues
- https://github.com/elastic/cloudbeat/pull/1376

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [x] Generate rule metadata using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

> **Note** Make sure to bump cloudbeat [`version/policy.go`](https://github.com/elastic/cloudbeat/blob/main/version/policy.go#L20) after merging this PR and drafting a new release.

### CODE REVIEW

File: bundle/compliance/cis_azure/rules/cis_9_4/test.rego [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import data.cis_azure.test_data
+import data.compliance.policy.azure.data_adapter
+import data.lib.test
+
+test_violation {
+	eval_fail with input as test_data.generate_azure_asset(
+		"azure-web-site",
+		{"clientCertMode": "NotRequired"},
+	)
```
Comment (orouz) [inline bundle/compliance/cis_azure/rules/cis_9_4/test.rego:11]: this is failing, not sure why yet

File: bundle/compliance/cis_azure/rules/cis_9_4/rule.rego [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+	# set result
+	result := common.generate_result_without_expected(
+		common.calculate_result(is_client_cert_mode_required),
+		data_adapter.resource,
+	)
+}
+
+is_client_cert_mode_required {
+	data_adapter.properties.clientCertMode == "Required"
```
Comment (oren-zohar) [inline bundle/compliance/cis_azure/rules/cis_9_4/rule.rego:18]: why are you querying `clientCertEnabled` and not `clientCertEnabled`?

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: bundle/compliance/cis_azure/rules/cis_9_4/rule.rego (hunk 2 above)
Comment (orouz) [reply bundle/compliance/cis_azure/rules/cis_9_4/rule.rego]: Azure portal audit mentions `clientCertEnabled` and azure cli audit mentions `ClientCertEnabled`. changed to the latter

Comment (oren-zohar) [reply bundle/compliance/cis_azure/rules/cis_9_4/rule.rego]: are both accessible in Azure Graph API?

File: bundle/compliance/policy/azure/data_adapter.rego [hunk 3]
```diff
@@ -17,3 +17,7 @@ is_storage_account {
 is_storage_account {
 	input.type == "azure-classic-storage-account"
 }
+
+is_website_asset {
+	input.type == "azure-web-site"
```
Comment (oren-zohar) [inline bundle/compliance/policy/azure/data_adapter.rego:22]: still no idea why you split it into two words lol

**Review by oren-zohar — APPROVED** [review]

File: bundle/compliance/policy/azure/data_adapter.rego (hunk 3 above)
Comment (orouz) [reply bundle/compliance/policy/azure/data_adapter.rego:22]: yeah just went with it. eventually we'll generate the `type/subtype` from the asset type's string. no need to invent new strings

**Pull request merged**